### PR TITLE
Fix #1127: correct question warning attribute

### LINKED ISSUE #1127: Questionnaire warning is broken
https://github.com/fsr-itse/EvaP/commit/a6f4cfab4d1fbac5881663dbb20f381fd28e3816#diff-bf2dc45ddddaa5eee703ff58c2f41020L3

in that line, `questionnaire_warning` was replaced by `section.questionnaire_warning`, it should be `question.warning`.

The fix is most likely trivial, but since this was broken for so long without anyone noticing, we need a test for this.

### PR DESCRIPTION
**Pull request by felixrindt**
See [issue](https://github.com/fsr-itse/EvaP/issues/1127). Will work on a regression test next.

### CODE REVIEW

**Comment by felixrindt** [comment]
Please review :)

File: evap/results/tests/test_views.py [hunk 1]
```diff
@@ -54,6 +54,13 @@ def setUpTestData(cls):
         mommy.make(Contribution, course=cls.course, contributor=responsible, can_edit=True, responsible=True, comment_visibility=Contribution.ALL_COMMENTS)
         mommy.make(Contribution, course=cls.course, contributor=contributor, can_edit=True)
 
+    def _create_questionaire_with_answers(self, answer_count=10):
+        questionnaire = mommy.make(Questionnaire)
+        likert_question = mommy.make(Question, type="L", questionnaire=questionnaire, order=2)
+        contributor = mommy.make(UserProfile)
+        contribution = mommy.make(Contribution, course=self.course, questionnaires=[questionnaire], contributor=contributor)
+        mommy.make(RatingAnswerCounter, question=likert_question, contribution=contribution, answer=3, count=answer_count)
```
[the remark this answers is not in the document]
Comment (karyon) [reply evap/results/tests/test_views.py]: ...actually, whether or not you use that, please try creating a new test class. we are trying to get away from these hard-to-maintain fixtures, and each new test here will make it harder to remove the fixture.

File: evap/results/tests/test_views.py [hunk 2]
```diff
@@ -73,6 +80,27 @@ def test_heading_question_filtering(self):
...
+        url = '/results/semester/%s/course/%s' % (self.semester.id, self.course.id)
+        user = 'evap'
+        page = self.get_assert_200(url, user)
+        self.assertNotIn("Only a few participants answered these questions.", page)
+
+    def test_few_answers_course_show_warning(self):
+        self._create_questionaire_with_answers(answer_count=3)
+        url = '/results/semester/%s/course/%s' % (self.semester.id, self.course.id)
+        user = 'evap'
+        page = self.get_assert_200(url, user)
```
Comment (karyon) [inline evap/results/tests/test_views.py]: you can put `user='evap'` into the parameter list directly, no need to introduce a variable for that.

File: evap/results/tests/test_views.py [hunk 3]
```diff
@@ -27,6 +27,36 @@ def setUpTestData(cls):
         cls.semester = mommy.make(Semester, id=1)
 
 
+class TestResultsViewContributionWarning(WebTest):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.semester = mommy.make(Semester, id=3)
+        mommy.make(UserProfile, username='contributor', groups=[Group.objects.get(name='Staff')], email="contributor@institution.example.com")
```
Comment (karyon) [inline evap/results/tests/test_views.py]: is that user supposed to be a contributor or staff user?

File: evap/results/tests/test_views.py [hunk 4]
```diff
@@ -27,6 +27,36 @@ def setUpTestData(cls):
...
+        contributor = UserProfile.objects.get(username="contributor")
```
Comment (karyon) [inline evap/results/tests/test_views.py]: merge these two lines to `contributor = mommy.make...`?

File: evap/results/templates/results_course_detail_questionnaires.html [hunk 5]
```diff
@@ -1,13 +1,13 @@
 {% load evaluation_filters %}
 
-<div class="inline-headers">
-    {% if section.questionnaire_warning and not contributor_data.total_votes == 0 %}
-        <p class="pull-right participants-warning questionnaire-warning">
+<div class="d-flex">
+    <h4>{{ section.questionnaire.public_name }}</h4>
+    {% if section.warning and not contributor_data.total_votes == 0 %}
+        <p class="ml-auto participants-warning questionnaire-warning">
```
Comment (janno42) [inline evap/results/templates/results_course_detail_questionnaires.html]: please also add `mt-auto`

File: evap/results/tests/test_views.py (hunk 3 above)
Comment (felixrindt) [reply evap/results/tests/test_views.py]: I split their personality.

File: evap/results/tests/test_views.py (hunk 4 above)
Comment (felixrindt) [reply evap/results/tests/test_views.py]: done

**Comment by karyon** [comment]
sorry, this fell off my radar!

**Pull request merged**